pyscripts/docker_log_processor.py

```diff
@@ -3,42 +3,64 @@
 # Licensed under the MIT license. See LICENSE file in the project root for
 # full license information.
 #
-# docker_log_processor.py
-# v-greach@microsoft.com
+# filename: docker_log_processor.py
+# author:   v-greach@microsoft.com
+# created:  01/29/2019
+# Rev: 03/03/2019 G
 
 from multiprocessing import Process, Queue, Event
 from threading import Thread
 from datetime import datetime, timedelta
 import docker
 import time
+import argparse
 import sys
 
 class DockerLogProcessor:
 
-    def __init__(self, container_names, options=""):
+    def __init__(self, args):
 
-        run_static = False
-        if options:
-            if "-staticfile" in options:
-                run_static = True
-                self.process_static_log(options)
+        # Parse args
+        parser = argparse.ArgumentParser(description="Docker Log Processor")
+        parser.add_argument('-staticfile', action='append', nargs='+', help="filename to read from")
+        parser.add_argument('-modulename', action='append', nargs='+', help="docker modulename to read from")
+        parser.add_argument('-filterfile', nargs=1, help="filename of json filters")
```

Comment:
Here's a neat trick. -staticfile and -modulename are mutually exclusive, so you can something like this and argparse will enforce the mutual exclusivity:
group = parser.add_mutually_exclusive_group()
group.add_argument('-staticfile', action='append', nargs='+', help="filename to read from")
group.add_argument('-modulename', action='append', nargs='+', help="docker modulename to read from")